### ISSUE TITLE
Greater than > in template function declaration not parsed correctly

### ISSUE THREAD

**Issue by tttapa** [doxygen/doxygen#8622]
Consider the following template function declaration and definition:
```cpp
#pragma once

template <bool>
struct S {};

template <int A, int B>
S<(A > B)> fun();

template <int A, int B>
S<(A > B)> fun() {
    return {};
}
```
When running Doxygen, this results in the following warning:
```text
header.hpp:10: warning: no matching file member found for 
S<(A >) B fun
Possible candidates:
 'S<(A > B)> fun()' at line 10 of file header.hpp
```

Interestingly enough, the generated documentation looks correct at first sight, and when removing either the declaration or the definition, Doxygen no longer complains.

**Expected behavior**
The template being parsed correctly and no warnings being emitted.

**To Reproduce**
Default Doxyfile with the following changes:
```diff
67c67
< EXTRACT_ALL            = NO
---
> EXTRACT_ALL            = YES
266c266
< GENERATE_LATEX         = YES
---
> GENERATE_LATEX         = NO
```
[Doxygen-8622.zip](https://github.com/doxygen/doxygen/files/6716455/Doxygen-8622.zip)

**Version**
`doxygen 1.9.2 (ff2a7670c50743547495f0ebee35323b11a74e6a)`, master branch, compiled from source just now, on Ubuntu 20.04.

**Comment by tttapa** [comment]
The function is just a free function with `S` as the return type, similar to how you would write non-templated function prototypes:

```cpp
// prototype
void fun();
// usage ...
void fun() {
   // implementation
}
```

**Comment by albert-github** [comment]
Ah yes now I see, I think that doxygen got a bit confused as well ...

**Comment by tttapa** [comment]
To clarify: this is by no means a high-priority bug, the spurious warnings just obscure some of the more important warnings.

**Comment by doxygen** [comment]
Please verify if the referenced commit fixes the issue.

**Comment by tttapa** [comment]
That seems to have fixed it, all such warnings have disappeared in my project. Thanks a lot for the quick responses!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
